### ISSUE TITLE
ReferencePolicy to support istio Gateway API

### ISSUE THREAD

**Issue by shonecyx** [istio/istio#45088]
Currently, only Secret has reference policy, and only implemented by Gateway API controller.

https://github.com/istio/istio/blob/master/pilot/pkg/model/push_context.go#L2145
```
func (ps *PushContext) ReferenceAllowed(kind config.GroupVersionKind, resourceName string, namespace string) bool {
	// Currently, only Secret has reference policy, and only implemented by Gateway API controller.
	switch kind {
	case gvk.Secret:
		if ps.GatewayAPIController != nil {
			return ps.GatewayAPIController.SecretAllowed(resourceName, namespace)
		}
	default:
	}
	return false
}

```
The istio Gateway api doesn't support ReferencePolicy/ReferenceGrant
https://github.com/istio/istio/blob/master/pilot/pkg/xds/sds.go#L271-L286

The challenge we are facing is that, our internal CA will generate the cert/key into the CA namespace and we are using the istio geteway api. Now we are working on the istio upgrade to 1.17 and got this challenge, SDS is blocked due to below error:

```
others: cross namespace secret reference requires ReferencePolicy
```

We would like to align with the gateway-api [security mode](https://gateway-api.sigs.k8s.io/concepts/security-model/) which does honor the namespace boundaries and use ReferencePolicy for cross namespace secret reference, however since we are not using gateway-api as of now this is not doable.

So I'd like to request this feature ReferencePolicy to support istio Gateway API

**Describe the feature request**

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Installation
[ x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
This isn't sufficient to provide secure cross-namespace credentials in Istio's model, since the Gateway and Deployment can cross namespace. So anyone can create a pod with label `istio=ingressgateway` and be implicitly granted access to the secret.

**Comment by howardjohn** [comment]
https://docs.google.com/document/d/1gDhj058q2PKKI9SWpxE23ExEXdOkLiueT8JA_oEJ8Mo/edit#heading=h.e52u9pac28y5

**Comment by shonecyx** [comment]
@howardjohn  Thanks for comment. Is there plan to support the cross namespace secret reference for Istio's model? Or we have to either move secret to ingressgateway namespace or migrate to the  kubernetes gateway-api to use ReferencePolicy?

**Comment by howardjohn** [comment]
No, I don't think that will ever be supported. The Istio model is fundamentally broken for that type of access IMO, as the doc I mentioned above discusses
